Formula/wine.rb

```diff
@@ -88,68 +85,43 @@ def require_universal_deps?
   # These libraries are not specified as dependencies, or not built as 32-bit:
   # configure: libv4l, gstreamer-0.10, libcapi20, libgsm
 
-  # Wine loads many libraries lazily using dlopen calls, so it needs these paths
-  # to be searched by dyld.
-  # Including /usr/lib because wine, as of 1.3.15, tries to dlopen
-  # libncurses.5.4.dylib, and fails to find it without the fallback path.
-
-  def library_path
-    paths = %W[#{HOMEBREW_PREFIX}/lib /usr/lib]
-    paths.unshift(MacOS::X11.lib) if build.with? "x11"
-    paths.join(":")
-  end
-
-  def wine_wrapper; <<-EOS.undent
-    #!/bin/sh
-    DYLD_FALLBACK_LIBRARY_PATH="#{library_path}" "#{bin}/wine.bin" "$@"
-    EOS
-  end
-
   def install
-    ENV.m32 # Build 32-bit; Wine doesn't support 64-bit host builds on macOS.
-
     # Help configure find libxml2 in an XCode only (no CLT) installation.
     ENV.libxml2
 
-    args = ["--prefix=#{prefix}"]
-    args << "--disable-win16" if MacOS.version <= :leopard
-    args << "--enable-win64" if build.with? "win64"
+    args32 = ["--prefix=#{prefix}"]
 
     # 64-bit builds of mpg123 are incompatible with 32-bit builds of Wine
-    args << "--without-mpg123" if Hardware::CPU.is_64_bit?
-
-    args << "--without-x" if build.without? "x11"
-
-    system "./configure", *args
-
-    # The Mac driver uses blocks and must be compiled with an Apple compiler
-    # even if the rest of Wine is built with A GNU compiler.
-    unless ENV.compiler == :clang || ENV.compiler == :llvm || ENV.compiler == :gcc
-      system "make", "dlls/winemac.drv/Makefile"
-      inreplace "dlls/winemac.drv/Makefile" do |s|
-        # We need to use the real compiler, not the superenv shim, which will exec the
-        # configured compiler no matter what name is used to invoke it.
-        cc = s.get_make_var("CC")
-        cxx = s.get_make_var("CXX")
-        s.change_make_var! "CC", cc.sub(ENV.cc, "xcrun clang") if cc
-        s.change_make_var! "CXX", cc.sub(ENV.cxx, "xcrun clang++") if cxx
-
-        # Emulate some things that superenv would normally handle for us
-        # Pass the sysroot to support Xcode-only systems
-        cflags  = s.get_make_var("CFLAGS")
-        cflags += " --sysroot=#{MacOS.sdk_path}"
-        s.change_make_var! "CFLAGS", cflags
-      end
+    args32 << "--without-mpg123"
+
+    args32 << "--without-x" if build.without? "x11"
+    args32 << "--with-wine64=../wine-64-build" if build.with? "win64"
+
+    args64 = ["--prefix=#{prefix}"]
+    args64 << "--enable-win64"
+
+    args64 << "--without-x" if build.without? "x11"
+
+    if build.with? "win64"
+      mkdir "wine-64-build"
+      cd "wine-64-build"
+
+      system "../configure", *args64
+
+      system "make", "install"
```

Comment:
Nice. I think it'd read a bit clearer to build `args64` inside the `if build.with? "win64"` and build `args32` after this. If you use the `mkdir do` block then you don't need the manual `cd`s.